src/image.rs

```diff
@@ -167,3 +167,49 @@ impl Image<Vec<u8>> {
     }
 }
 
+/// A yuv image with pixels of type `T`.
+///
+/// Two variants of this type are commonly used:
+///
+/// - `YUVImage<&mut [u8]>`: mutable reference to yuv image data (output image for decompression by
+/// [`Decompressor`][crate::Decompressor]).
+/// - `YUVImage<Vec<u8>>`: owned yuv image data (you can convert it to a reference using
+/// [`.as_deref()`][YUVImage::as_deref] or [`.as_deref_mut()`][YUVImage::as_deref_mut]).
+pub struct YUVImage<T> {
+    /// Pixel data of the image (typically `&[u8]`, `&mut [u8]` or `Vec<u8>`).
+    pub pixels: T,
+    /// Width of the image in pixels (number of columns).
+    pub width: usize,
+    /// Pad the width of each line in each plane of the YUV image will be
+    /// padded to the nearest multiple of this number of bytes (must be a power of
+    /// 2.)  To generate images suitable for X Video, <tt>pad</tt> should be set to 4
+    pub pad: usize,
```

Comment:
TurboJPEG calls this value `align`, which is perhaps a better name, because it specifies the alignment, not the amount of padding.

Also, the first sentence of the comment could be improved, perhaps something like this would work (adapted from TurboJPEG docs):

> Row alignment (in bytes) of the image (must be a power of 2). Each row in each plane of the YUV image will be padded to the nearest multiple of `align`.

I would also drop the reference to "X Video": I'm not sure what it refers to, and googling this phrase finds only some sites with adult content.

Reply:
Ok, I basically stole this from your code in `turbojpeg-sys/binding.rs` in `tjDecompressToYUV2`. I do in fact see that the newer [tj3DecompressToYUV8](https://github.com/libjpeg-turbo/libjpeg-turbo/blob/ebca79d508d424687707462675a4a5f09ab4429c/turbojpeg.h#L1678) does have the `align` naming. It also still mentions the `X Video` part. Please advise, but I'd like to keep the lib naming matching the function binding.

Reply:
Weird, it seems that this parameter was named `pad `[in TurboJPEG 1.5.x](https://rawcdn.githack.com/libjpeg-turbo/libjpeg-turbo/1.5.x/doc/html/group___turbo_j_p_e_g.html#ga39e08906528db5a764670ea48d344b09), but was renamed to `align` [in TurboJPEG 2.0.x](https://rawcdn.githack.com/libjpeg-turbo/libjpeg-turbo/2.0.x/doc/html/group___turbo_j_p_e_g.html#ga5a3093e325598c17a9f004323af6fafa). However, for some reason, the `turbojpeg.h` that is bundled with this crate is from version 2.1.2, but still uses `pad`!

In any case, the `align` name still seems preferable, because it's used by latest versions of TurboJPEG and it better describes the intent.

Reply:
Ok, align it is. Just copied the documentation from that 2.0.x link you shared.